### ISSUE TITLE
DeleteMessageBatch Issue - Top level element may not be treated as a list 

### ISSUE THREAD

**Issue by onema** [aws/aws-sdk-php#637]
I'm currently migrating a small application using the AWS SDK for PHP from `v2` to `v3`. The transition has been smooth, but the  `DeleteMessageBatch` is throwing and exception with the  following message:

```
Error executing "DeleteMessageBatch" on "<url remooved>"; 
    AWS HTTP error: 
    Client error: 
    400 MalformedInput (client): 
    Top level element may not be treated as a list
```
```xml
<?xml version="1.0"?>
<ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
    <Error>
        <Type>Sender</Type>
        <Code>MalformedInput</Code>
        <Message>Top level element may not be treated as a list</Message>
        <Detail/>
    </Error>
    <RequestId>[request id removed]</RequestId>
</ErrorResponse>
```
parameters are set just like the [docs](http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-sqs-2012-11-05.html#deletemessagebatch).

Similar issues have been logged here:
 - [SQS DeleteMessageBatch Errors #161](https://github.com/aws/aws-sdk-go/issues/161)
 - [ SQS DeleteMessageBatch Errors #630 ](https://github.com/aws/aws-sdk-php/issues/630)

Please let me know if there is any additional information.

**Comment by jeskew** [comment]
Thanks for bringing this to our attention. We are working on a response.

**Comment by jeskew** [comment]
PR #638 has been merged into master and should fix this.

I'm going to go ahead and close the issue, but please feel to reopen it if you have any questions.

**Comment by allen-infante** [comment]
Hi, when migrated same  issue.

http://docs.aws.amazon.com/aws-sdk-php/v3/api/api-sqs-2012-11-05.html#deletemessagebatch

$result = $client->deleteMessageBatch([
    '**Entries**' => [ // REQUIRED
        [
            'Id' => '<string>', // REQUIRED
            'ReceiptHandle' => '<string>', // REQUIRED
        ],
        // ...
    ],
    'QueueUrl' => '<string>', // REQUIRED
]);
was working with previous version and with these parameter, now throw error :

Error executing "DeleteMessageBatch" on "https://XXXXX";
AWS HTTP error: Client error: `POST https://XXXXX` resulted in a `400 Bad Request` response:
<?xml version="1.0"?><ErrorResponse xmlns="http://queue.amazonaws.com/doc/2012-11-05/">
<Error><Type>Sender</Type><Code>MalformedInput</Code><Message>Top level element may not be treated as a list</Message><Detail/></Error><RequestId>68e958b1-c421-5ae2-a345-17c09b1ecd0a</RequestId></ErrorResponse>

if debug url generated is like this : 

&Action=DeleteMessageBatch
&Version=2012-11-05
&**Entries**.1.Id=XXX&
**Entries**.1.ReceiptHandle=XXX

http://docs.aws.amazon.com/en_us/AWSSimpleQueueService/latest/APIReference/API_DeleteMessageBatch.html

but sample display in this way :

&Action=DeleteMessageBatch
&Version=2012-11-05
&**DeleteMessageBatchRequestEntry**.1.Id=msg1
&**DeleteMessageBatchRequestEntry**.1.ReceiptHandle=

does this matter?

Regards

**Comment by jeskew** [comment]
@allen-infante Could you try updating to a later release of the SDK? The issue you're describing was addressed in #638.
